cmd/lncli/walletrpc_active.go

```diff
@@ -396,3 +401,200 @@ func labelTransaction(ctx *cli.Context) error {
 
 	return nil
 }
+
+// fundPsbtResponse is a struct that contains JSOn annotations for nice result
+// serialization.
+type fundPsbtResponse struct {
+	Psbt  string                 `json:"psbt"`
+	Locks []*walletrpc.UtxoLease `json:"locks"`
+}
+
+var fundPsbtCommand = cli.Command{
+	Name:      "fundpsbt",
+	Category:  "On-chain",
+	Usage:     "Fund a Partially Signed Bitcoin Transaction (PSBT).",
+	ArgsUsage: "psbt",
+	Description: `
+	The fundpsbt command creates a fully populated PSBT that contains enough
+	inputs to fund the outputs specified in the template PSBT.
+
+	If there are no inputs specified in the template, coin selection is
+	performed automatically. If the template does contain any inputs, it is
+	assumed that full coin selection happened externally and no additional
+	inputs are added. If the specified inputs aren't enough to fund the
+	outputs with the given fee rate, an error is returned.
+
+	After either selecting or verifying the inputs, all input UTXOs are
+	locked with an internal app ID.
+	`,
+	Flags: []cli.Flag{
+		cli.Uint64Flag{
+			Name: "conf_target",
+			Usage: "the number of blocks that the transaction " +
+				"should be confirmed on-chain within",
+			Value: 6,
+		},
+		cli.Uint64Flag{
+			Name: "sat_per_vbyte",
+			Usage: "a manual fee expressed in sat/vbyte that " +
+				"should be used when creating the transaction",
+		},
+	},
+	Action: actionDecorator(fundPsbt),
+}
+
+func fundPsbt(ctx *cli.Context) error {
+	// Display the command's help message if we do not have the expected
+	// number of arguments/flags.
+	if ctx.NArg() != 1 {
+		return cli.ShowCommandHelp(ctx, "fundpsbt")
+	}
+
+	walletClient, cleanUp := getWalletClient(ctx)
+	defer cleanUp()
+
+	args := ctx.Args()
+	psbtBytes, err := base64.StdEncoding.DecodeString(args.First())
```

Comment:
Not super clear from the help or description that the first value is the raw PSBT template, this should also be a named argument as well IMO.